### PR TITLE
Fix openai error handler not working for async prompty

### PR DESCRIPTION
**Pull request by ninghu**
# Description

This PR includes two changes:

1. During tests, we discovered that the OpenAI error handler was not working for async prompty. Async version error handler has been added to resolve this issue.
2. Expose the timeout for LLM call. This will be utilized by the evaluators in a later PR.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/microsoft/promptflow/blob/main/CONTRIBUTING.md).**
- [ ] **I confirm that all new dependencies are compatible with the MIT license.**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-core/promptflow/core/_prompty_utils.py [hunk 1]
```diff
@@ -1162,6 +1163,114 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
+def handle_openai_error_async(tries: int = 10, unprocessable_entity_error_tries: int = 3):
```
Comment (ninghu) [inline src/promptflow-core/promptflow/core/_prompty_utils.py:1166]: I tried to merge this code with the existing handle_openai_error decorator, but found it difficult. So, I created a separate decorator for the async version to keep the code clean

File: src/promptflow-core/promptflow/core/_flow.py [hunk 2]
```diff
@@ -556,7 +558,8 @@ async def __call__(self, *args, **kwargs) -> Mapping[str, Any]:
         # 4. send request to OpenAI
         api_client = get_open_ai_client_by_connection(connection=connection, is_async=True)
 
-        response = await send_request_to_llm(api_client, self._model.api, params)
+        timeout = kwargs.get("timeout", None)
```
Comment (ninghu) [inline src/promptflow-core/promptflow/core/_flow.py:561]: This will be consumed by evaluators in my next PR.

**Review by lalala123123 — APPROVED** [review]

File: src/promptflow-core/promptflow/core/_prompty_utils.py [hunk 3]
```diff
@@ -192,11 +193,11 @@ def get_open_ai_client_by_connection(connection, is_async=False):
     return client
 
 
-def send_request_to_llm(client, api, parameters):
+def send_request_to_llm(client, api, parameters, timeout):
     if api == "completion":
-        result = client.completions.create(**parameters)
+        result = client.with_options(timeout=timeout).completions.create(**parameters)
```
Comment (lalala123123) [inline src/promptflow-core/promptflow/core/_prompty_utils.py:198]: Hi @ninghu , why the timeout needs to be configured with with_options? timeout can be also configured by create function passed by parameters.
image

Comment (ninghu) [reply src/promptflow-core/promptflow/core/_prompty_utils.py:198]: The change is by referencing the openai documentation here: https://github.com/openai/openai-python#timeouts

image

**Review by luigiw — APPROVED** [review]

**Review by lalala123123 — APPROVED** [review]

**Pull request merged**